types/result.go

```diff
@@ -12,6 +12,7 @@ type Result struct {
 	Code CodeType   `json:"code"`
 	Data data.Bytes `json:"data"`
 	Log  string     `json:"log"` // Can be non-deterministic
+	Tags []*KVPair  `json:"tags"`
```

Comment:
I think we should start separating the result from DeliverTx and CheckTx.
DeliverTx needs tags, which CheckTx doesn't.
We will also need to add Gas and Fees (or Cost and Value or....) to CheckTx as per #57 and soon.

Given this, I already separated the DeliverTx and CheckTx results in the sdk. We should use different types in the abci.